### DISCUSSION TITLE
v5.1.0にバージョンアップ後、SAML認証を用いて新規ユーザ登録できない

### DISCUSSION THREAD

**Discussion by hiroshigenba** [growilabs/growi#7692] [Help / Questions] [answered]
Environment
------------

### Host

| item     | version |
| ---      | --- |
|OS        |CentOS 7|
|GROWI     |5.1.0|
|node.js   |16.16.0|
|npm       |8.11.0|
|Using Docker|yes|
|Using [growi-docker-compose][growi-docker-compose]|yes|

[growi-docker-compose]: https://github.com/weseek/growi-docker-compose

*(Accessing https://{GROWI_HOST}/admin helps you to fill in above versions)*

### Client

| item     | version |
| ---      | --- |
|OS        |Windows10|
|browser   |Google Chrome 103.0|

How to reproduce? (再現手順)
---------------------------
SAML認証を用いたユーザ作成を実施

What happens? (症状)
---------------------
1. 認証はSAML認証を用いており、新規ユーザを作成したところ、新規ユーザの初回設定画面に遷移しない。
    1. IdP でログイン
    2. GROWI `/login` に帰って来る
        -  LoginForm の warning alert 表示部分に以下のエラーメッセージが出る
            - Username should not be null. Please check Authentication Mechanism Settings on admin page
4. 管理画面上ではブランクのユーザが作成されてログインができなかった模様。
5. Basic認証でログインできるよう新規ユーザを作成したところ実施できた

What is the expected result? (期待される動作)
-------------------------------------------
以前のバージョン同様にSAML認証で新規ユーザを作成した場合に問題なく作成できる

**Comment by yuki-takei** [comment]
Introduced by https://github.com/weseek/growi/commit/3cb1df4f6968523307e850826436a3acddb77b5f

**Comment by kaori-tokashiki** [comment]
@h-genba-w 
Issueの起票ありがとうございます。

saml 設定の Attribute Mapping で ユーザー名に何も入力していない場合、ユーザーの新規登録時に上記のエラーが表示されることを確認いたしました。


1. 管理画面 > セキュリティ設定 (/admin/security) に遷移
2. 認証機構設定 で saml のタブを押す
3. saml 設定の Attribute Mapping で ユーザー名 の項目を見る
ユーザー名に入力されているか確認していただけますでしょうか?

~~ユーザー名に何も設定されていない場合~~
~~1. username と入力~~
~~2. 更新ボタンを押す~~
~~で、新規ユーザ登録できるか確認していただけますでしょうか?~~


## Memo
影響のある変更PR
https://github.com/weseek/growi/pull/6019/files

**Comment by kaori-tokashiki** [comment, marked as answer]
@h-genba-w

すみません、上記でコメントした内容は間違っていました。

**saml設定のAttribute Mappingでユーザー名の値がセットされているか**確認お願いします。

手順は下記の通りです。
1. 管理画面 > セキュリティ設定 (/admin/security) に遷移
2. 認証機構設定 で saml のタブを押す
3. saml 設定の Attribute Mapping 

<img width="1236" alt="Screen Shot 2022-09-22 at 20 34 53" src="https://user-images.githubusercontent.com/59536731/191736314-e943b51e-3c71-4cb5-9ab1-8c3466f090a4.png">

## ユーザー名に値が入っていない場合
添付写真のようにユーザー名に値が入っていないといった旨のアラートは表示されていますでしょうか? 
表示されている場合、値を入力して更新お願いします。

※ もしも、アラートが表示されていない場合は、こちら側で改修が必要になります。

<img width="877" alt="Screen Shot 2022-09-22 at 13 44 17" src="https://user-images.githubusercontent.com/59536731/191736640-478c5267-b402-4b74-ae51-f709c5ad8343.png">

## ユーザー名に値が入っている場合
その値が間違っているので、設定値の見直しをお願いします。

## 参考URL
https://teachme.jp/35563/manuals/16518665

**Comment by kaori-tokashiki** [comment]
#6631 の修正とは関係がないため reopen
